internal/pkg/levee/levee_ear_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package levee
+
+import (
+	"testing"
+
+	"github.com/google/go-flow-levee/internal/pkg/debug"
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+func TestLeveeEAR(t *testing.T) {
+	dataDir := analysistest.TestData()
+	if err := Analyzer.Flags.Set("config", dataDir+"/test-config.yaml"); err != nil {
+		t.Error(err)
+	}
+	if err := Analyzer.Flags.Set("useEAR", "true"); err != nil {
+		t.Error(err)
+	}
+	if *debugging {
+		Analyzer.Requires = append(Analyzer.Requires, debug.Analyzer)
+	}
+
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/arguments")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/basictypes")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/binop")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/booleans")
+	// analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/callorder")  // TODO: flow-sensitive
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/closures")
+	// analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/collections") // TODO: map key
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/declarations")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/eface")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/embedding")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/excludedpackage")
+	analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/extracts")
+	// analysistest.Run(t, dataDir, Analyzer, "./src/levee_analysistest/example/tests/fields")  // TODO: FP has been fixed?
```

Comment:
It seems there are also some false negatives here, e.g. the return value of `s.GetData()` where `s` has type `core.Source` should be tainted.

Reply:
s.GetData() is defined in another package "core" so its body is not available to the EAR analysis yet. This will be handled in the future with a proper way to link packages.